src/ansys/tools/versioning/utils.py

```diff
@@ -276,3 +281,122 @@ def wrapper(self, *args, **kwargs):
         return wrapper
 
     return decorator
+
+
+class Version(int):
+    """Class for version comparison.
+
+    This class can be instantiated from a string or an integer.
+    The constructor will choose the corresponding class.
+
+    Examples
+    --------
+    >>> from ansys.tools.versioning.utils import Version
+    >>> Version(1)
+    1
+    >>> Version("dev")
+    'dev'
+    >>> Version(1) <= Version("dev")
+    True
+    >>> Version(99999) >= Version("dev")
+    False
+    """
+
+    def __new__(cls, value):
```

Comment:
```suggestion
    def __new__(cls, value : Union[str, int]):
```
This allows when type hinting both arguments. Per definition of your ``__new__`` method, it should look like this.